mm2src/coins/utxo/utxo_common.rs

```diff
@@ -1006,37 +1027,52 @@ pub async fn p2sh_spending_tx<T: UtxoCommonOps>(coin: &T, input: P2SHSpendingTxI
         )
         .await
     );
-    let signed_input = try_s!(p2sh_spend(
-        &unsigned,
-        DEFAULT_SWAP_VOUT,
-        input.keypair,
-        input.script_data,
-        input.redeem_script.into(),
-        coin.as_ref().conf.signature_version,
-        coin.as_ref().conf.fork_id
-    ));
-    Ok(UtxoTx {
-        version: unsigned.version,
-        n_time: unsigned.n_time,
-        overwintered: unsigned.overwintered,
-        lock_time: unsigned.lock_time,
-        inputs: vec![signed_input],
-        outputs: unsigned.outputs,
-        expiry_height: unsigned.expiry_height,
-        join_splits: vec![],
-        shielded_spends: vec![],
-        shielded_outputs: vec![],
-        value_balance: 0,
-        version_group_id: coin.as_ref().conf.version_group_id,
-        binding_sig: H512::default(),
-        join_split_sig: H512::default(),
-        join_split_pubkey: H256::default(),
-        zcash: coin.as_ref().conf.zcash,
-        posv: coin.as_ref().conf.is_posv,
-        str_d_zeel: unsigned.str_d_zeel,
-        tx_hash_algo: unsigned.hash_algo.into(),
-        v_extra_payload: None,
-    })
+
+    match input.signer {
+        P2SHSigner::KeyPair(key_pair) => {
+            let signed_input = try_s!(p2sh_spend(
+                &unsigned,
+                DEFAULT_SWAP_VOUT,
+                &key_pair,
+                input.script_data,
+                input.redeem_script.into(),
+                coin.as_ref().conf.signature_version,
+                coin.as_ref().conf.fork_id
+            ));
+            Ok(UtxoTx {
```

Comment:
I can see there is a `complete_tx` function which creates UtxoTx. Could it be used here?

Reply:
[86e3b2b](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2566/commits/86e3b2bf60579563daa90df3a061de6fcf2b1979)